core/src/test/java/org/elasticsearch/index/query/QueryParseContextTests.java

```diff
@@ -89,6 +109,10 @@ public void testParseInnerQueryBuilderEmptyBody() throws IOException {
             QueryParseContext context = new QueryParseContext(indicesQueriesRegistry, parser, ParseFieldMatcher.EMPTY);
             Optional<QueryBuilder> emptyQuery = context.parseInnerQueryBuilder();
             assertFalse(emptyQuery.isPresent());
+            // regardless of strict mode, we should get warnings in the response headers
```

Comment:
remove this comment? strict mode will be gone soon, one less edit needed later :)